packages/query-core/src/queryObserver.ts

```diff
@@ -582,6 +582,11 @@ export class QueryObserver<
       isRefetchError: isError && hasData,
       isStale: isStale(query, options),
       refetch: this.refetch,
+      promise: (prevQuery.promise ??
+        query.promise ??
+        new Promise(() => {
+          // never resolves? 🤷‍♂️
+        })) as any,
```

Comment:
so, I know we are supposed to cache the promise, but is this also true for immediately resolved promises? Like, I'm thinking: If we have `data` (`hasData`), then we should _always_ immediately resolve with it. There are too many cases where this could be set, like with `keepPreviousData` or when we have stale data, or `initialData`, ...

So the only time where we want to integrate with suspense / error boundaries is on hard load. So would this work?

```
promise: hasData
  ? Promise.resolve(data)
  : (query.promise ?? new Promise(() => {})),
```

Reply:
Yeah, the promise should be a stable reference to `Promise.resolve(data)` when it's already resolved

I just wanted to see if I could prototype get _anything_ working, knowingly it was wrong. Didn't have much time for it though